lxd/storage/drivers/driver_pure_util.go

```diff
@@ -12,3 +49,186 @@ func newPureClient(driver *pure) *pureClient {
 		driver: driver,
 	}
 }
+
+// createBodyReader creates a reader for the given request body contents.
+func (p *pureClient) createBodyReader(contents map[string]any) (io.Reader, error) {
+	body := &bytes.Buffer{}
+
+	err := json.NewEncoder(body).Encode(contents)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to write request body: %w", err)
+	}
+
+	return body, nil
+}
+
+// request issues a HTTP request against the PureStorage gateway.
+func (p *pureClient) request(method string, path string, reqBody io.Reader, reqHeaders map[string]string, respBody any, respHeaders map[string]string) error {
+	var url string
+
+	// Construct the request URL.
+	if strings.HasPrefix(path, "/api") {
```

Comment:
why are we using this function in such a way that the function needs to translate the path into what we actually want? This could do with a comment explaining i think.

Reply:
Yep, will include the comment. 

The `/api` paths do not include API version in front of them. This way, if `/api/...` is provided, we literally use that path, otherwise we prepend the required API version. 

Also to point out, currently the latest API version is always used (list of supported versions is retrieved from PureStorage), but could be something worth limiting or exposing through the config in the future.

Reply:
we should be targeting the API version we support right? Not using the latest as otherwise something might change and the pure appliance would expect us to use that newer version not the older one. This should be documented too.

Reply:
> The `/api` paths do not include API version in front of them

What are the "/api" paths? Given that in the else statement you then add /api/version to the path provided, it seems that infact some `/api` paths do have a version?

Reply:
Just `/api/api_versions`

Reply:
So should we do an exact match then?

Reply:
That will work as well. Will fix.